azure-webapp-maven-plugin/src/main/resources/schema/WebAppConfiguration.json

```diff
@@ -0,0 +1,291 @@
+        },
+        "configurationSource": {
+          "$ref": "#/definitions/non-empty-string"
+        }
+      },
+      "required": [
+        "name"
+      ]
+    },
+    "runtime": {
+      "title": "Runtime",
+      "description": "Runtime configuration for Maven plugin for Azure Web App",
+      "type": "object",
+      "properties": {
+        "os": {
+          "description": "The operating system for app service",
+          "type": "string",
+          "pattern": "(?i)^(windows|linux|docker)$"
+        },
+        "webContainer": {
+          "$ref": "#/definitions/non-empty-string"
+        },
+        "javaVersion": {
+          "$ref": "#/definitions/non-empty-string"
+        },
+        "image": {
+          "$ref": "#/definitions/non-empty-string"
+        },
+        "serverId": {
+          "$ref": "#/definitions/non-empty-string"
+        },
+        "registryUrl": {
+          "$ref": "#/definitions/non-empty-string"
+        }
+      },
+      "dependencies": {
+        "serverId": [
+          "image"
+        ],
+        "registryUrl": [
+          "image"
+        ]
+      },
+      "required": [
+        "os"
+      ],
+      "allOf": [
+        {
+          "if": {
+            "properties": {
+              "os": {
+                "pattern": "(?i)^(windows|linux)$"
+              }
+            }
+          },
+          "then": {
+            "required": [
+              "javaVersion",
+              "webContainer"
+            ]
+          }
+        },
+        {
+          "if": {
+            "properties": {
+              "os": {
+                "pattern": "(?i)^(docker)$"
+              }
+            }
+          },
+          "then": {
+            "required": [
+              "image"
+            ]
+          }
+        }
+      ]
+    },
+    "auth": {
+      "title": "AuthConfiguration",
+      "description": "The auth config for accessing azure resources",
+      "type": "object",
+      "properties": {
+        "type": {
+          "$ref": "#/definitions/auth-type"
+        },
+        "serverId": {
+          "description": "The server id defined in maven settings",
+          "type": "string"
+        },
+        "client": {
+          "description": "Client ID",
+          "type": "string"
+        },
+        "tenant": {
+          "description": "Tenant ID",
+          "type": "string"
+        },
+        "key": {
+          "description": "Password",
+          "type": "string"
+        },
+        "certificate": {
+          "description": "The absolute path of your certificate",
+          "type": "string"
+        },
+        "certificatePassword": {
+          "description": "The password for your certificate, if there is any",
+          "type": "string"
+        },
+        "environment": {
+          "description": "The Azure cloud environment",
+          "type": "string",
+          "default": "AZURE",
+          "pattern": "(?i)^(AZURE|AZURE_CHINA|AZURE_GERMANY|AZURE_US_GOVERNMENT)$"
```

Comment:
multiple format of azure environment.

Reply:
follow document here https://github.com/microsoft/azure-maven-plugins/wiki/Authentication#service-principles-in-plugin-configuration